main/main.cpp

```diff
@@ -1679,6 +1679,9 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 		driver_hints = "vulkan";
 #endif
 
+		//Set default Vulkan gpu to run, -1 will be default
+		GLOBAL_DEF("rendering/rendering_device/gpu_index", -1);
```

Comment:
```suggestion
		GLOBAL_DEF_RST("rendering/rendering_device/gpu_index", -1);
```
To force restart on changing it, as the docs already tell you to